lib/rust/lexer/definition/src/token.rs

```diff
@@ -195,12 +315,14 @@ pub enum LineEnding {
 }
 
 impl LineEnding {
+    const NO_LENGTH:usize = 0;
+
     /// Get the number of rust `char`s that the newline type takes up.
     pub fn size(self) -> usize {
         match self {
-            Self::None => 0,
-            Self::LF   => 1,
-            Self::CRLF => 2,
+            Self::None => Self::NO_LENGTH,
+            Self::LF   => lexeme::lf().length(),
+            Self::CRLF => lexeme::cr().length() + lexeme::lf().length(),
```

Comment:
I still dont like it. Its very very minor thing, but cant we just do `lexeme::crlf().length()` instead please?